tests/net/6lo/src/main.c

```diff
@@ -851,59 +852,47 @@ static struct net_6lo_data test_data_25 = {
 
 #endif
 
-static int test_6lo(struct net_6lo_data *data)
+static void test_6lo(struct net_6lo_data *data)
 {
 	struct net_pkt *pkt;
-	int result = TC_FAIL;
 
 	pkt = create_pkt(data);
-	if (!pkt) {
-		TC_PRINT("%s: failed to create buffer\n", __func__);
-		goto end;
-	}
-
+	zassert_not_null(pkt, "failed to create buffer");
 #if DEBUG > 0
 	TC_PRINT("length before compression %zu\n",
 		 net_pkt_get_len(pkt));
 	net_hexdump_frags("before-compression", pkt, false);
 #endif
 
-	if (!net_6lo_compress(pkt, data->iphc, NULL)) {
-		TC_PRINT("compression failed\n");
-		goto end;
-	}
+	zassert_true(net_6lo_compress(pkt, data->iphc, NULL),
+		     "compression failed");
 
 #if DEBUG > 0
 	TC_PRINT("length after compression %zu\n",
 		 net_pkt_get_len(pkt));
 	net_hexdump_frags("after-compression", pkt, false);
 #endif
 
-	if (!net_6lo_uncompress(pkt)) {
-		TC_PRINT("uncompression failed\n");
-		goto end;
-	}
-
+	zassert_true(net_6lo_uncompress(pkt),
+		     "uncompression failed");
 #if DEBUG > 0
 	TC_PRINT("length after uncompression %zu\n",
 	       net_pkt_get_len(pkt));
 	net_hexdump_frags("after-uncompression", pkt, false);
 #endif
 
 	if (compare_data(pkt, data)) {
-		result = TC_PASS;
+		zassert_true(1, NULL);
```

Comment:
put the compare_data() test into zassert() so same way as other part of these tests

Reply:
Fixed